tests/chainer_tests/functions_tests/math_tests/test_det.py

```diff
@@ -0,0 +1,236 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions as F
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.utils import type_check
+
+
+class DetFunctionTestBase(object):
+
+    def setUp(self):
+        self.x, self.y, self.gy = self.make_data()
+        self.ct = numpy.array(
+            [ix.T for ix in self.x.copy()]).astype(numpy.float32)
+
+    def det_transpose(self, gpu=False):
+        if gpu:
+            cx = cuda.to_gpu(self.x.copy())
```

Comment:
We do not have to copy GPU array. `self.setUp` is called and `self.x` is initialized every time we invoke each test fixture (whose name begins with `test`).